gridstatus/isone.py

```diff
@@ -227,6 +228,98 @@ def get_load_forecast(self, date, end=None, verbose=False):
+
+        Forecast is published for 7 days and generated daily by 10 am.
+        https://www.iso-ne.com/isoexpress/web/reports/operations/-/tree/seven-day-solar-power-forecast
+        """
+        return (
+            self._get_solar_or_wind_forecast(
+                date,
+                end,
+                resource_type="Solar",
+                verbose=verbose,
+            )
+            .reset_index(drop=True)
+            .sort_values(["Interval Start", "Publish Time"])
+        )
+
+    def get_wind_forecast(self, date, end=None, verbose=False):
+        """Return wind forecast published on a specific date
+
+        Forecast is published for 7 days and generated daily by 10 am.
+        https://www.iso-ne.com/isoexpress/web/reports/operations/-/tree/seven-day-wind-power-forecast
+        """
+        return (
+            self._get_solar_or_wind_forecast(
+                date,
+                end,
+                resource_type="Wind",
+                verbose=verbose,
+            )
+            .reset_index(drop=True)
+            .sort_values(["Interval Start", "Publish Time"])
+        )
+
+    @support_date_range(frequency="DAY_START")
+    def _get_solar_or_wind_forecast(
+        self,
+        date,
+        end=None,
+        resource_type="Wind",
+        verbose=False,
+    ):
+        """Return solar or wind forecast published on a specific date
+
+        Resource type can be "Solar" or "Wind"
+        """
+        if date == "latest":
+            date = pd.Timestamp.now(tz=self.default_timezone)
+
+        value_name = f"{resource_type.capitalize()} Forecast"
+
+        url = (
+            f"https://www.iso-ne.com/transform/csv/wphf?start={date.strftime('%Y%m%d')}"
+        )
+
+        # We return the raw data so we can extract metadata (publish time)
+        raw_string = _make_request(url, skiprows=None, verbose=verbose, return_raw=True)
+
+        # Remove lines that don't contain data table
+        cleaned_data = "\n".join(
+            [line for line in raw_string.split("\n") if line.startswith('"D"')],
+        )
+
+        # Create a buffer and read using pandas
+        df = pd.read_csv(StringIO(cleaned_data), header=None).drop(columns=[0, 1])
+
+        df.columns = df.iloc[1]
+        df = df.drop(index=[0, 1]).reset_index(drop=True)
+
+        data = df.melt(id_vars=["Hour Ending"], var_name="Date", value_name=value_name)
+
+        data["Interval Start"] = pd.to_datetime(
+            data["Date"] + " "
+            # Subtract 1 from the hour ending to get the start of the interval
+            # Make sure to 0-pad the hour
+            + (data["Hour Ending"].astype(int) - 1).astype(str).str.zfill(2) + ":00",
+            # TODO: might need to handle DST transitions
+        ).dt.tz_localize(self.default_timezone)
+
+        data["Interval End"] = data["Interval Start"] + pd.Timedelta(hours=1)
+
+        # Website says report is generally available by 10 am.
```

Comment:
The "Publish Time" comes from the webpage that says ([solar](https://www.iso-ne.com/isoexpress/web/reports/operations/-/tree/seven-day-solar-power-forecast), [wind](https://www.iso-ne.com/isoexpress/web/reports/operations/-/tree/seven-day-wind-power-forecast)):

> Download the aggregate wind/solar power forecast for each hour for the next seven days. The forecast is automatically updated daily, generally by 10:00 a.m. ET.